cmd/utils/flags.go

```diff
@@ -1634,9 +1634,11 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 	}
 	// Read the value from the flag no matter if it's set or not.
 	cfg.Preimages = ctx.Bool(CachePreimagesFlag.Name)
-	if cfg.NoPruning && !cfg.Preimages {
+	// Only check the user's input for hash mode here; path is the default scheme,
+	// so we don't need to check the actual database scheme.
+	if cfg.NoPruning && !cfg.Preimages && ctx.String(StateSchemeFlag.Name) == rawdb.HashScheme {
```

Comment:
Could you please revert this change for now? I still prefer to enable the preimage for archive node.

Reply:
Preimage is needed for both hash and path mode archive in order to traverse the states.

Reply:
I see, the dump state and some other related APIs need the preimage